core/tee/tee_svc_storage.c

```diff
@@ -710,7 +710,13 @@ TEE_Result syscall_storage_start_enum(unsigned long obj_enum,
 		return TEE_ERROR_ITEM_NOT_FOUND;
 
 	e->fops = fops;
-	assert(!e->dir);
+
+	if (e->fops && e->dir) {
+		e->fops->closedir(e->dir);
+		e->fops = NULL;
```

Comment:
Does not look correct (not set anywhere before returning).

Reply:
I think you're right, I'll remove this.